Ref/bug/save as

MichiGandi opened the pull request:
Fix the scene id problem on the save as dialog.

MichiGandi committed:
fix: scnee duplicate id

When used save scene as, the new scene got the same id as the old scene.

MichiGandi committed:
fix: allow overriding scenes

It was not possible to override scenes. The save file dialog would just open the scene folder and save the new scene inside. This overrides the scene now.

MGH-Automator commented:
Task linked: [CU-861m92tkm fix: save as does save duplicate guid](https://app.clickup.com/t/861m92tkm)

cblech commented on Assets/Scripts/EditorState/SceneManagerState.cs:
```diff
@@ -4,6 +4,7 @@
 using System.Linq;
 using JetBrains.Annotations;
 using Assets.Scripts.Utility;
+using System.Data;
```
Using directive is unnecessary

cblech commented on Assets/Scripts/System/SceneManagerSystem.cs:
```diff
@@ -5,6 +5,7 @@
 using System;
 using System.IO;
 using System.Linq;
+using UnityEditor;
```
Using directive is unnecessary
And will throw errors when building standalone

cblech commented on Assets/Scripts/System/SceneManagerSystem.cs:
```diff
@@ -243,19 +244,23 @@ private void SaveSceneAs(SceneDirectoryState sceneDirectoryState)
+            // Strip path inside a dcl scene folder. This allows to save as a already existing scene.
+            newPath = newPath.Substring(0, newPath.IndexOf(".dclscene") + 9);
+
+            Guid newId = default;
```
Use `Guid.Empty` to be more explicit

MichiGandi committed:
fix requests

cblech reviewed (changes requested):
There is a different problem now. When "Saving as..." this scene to a new location
![grafik](https://user-images.githubusercontent.com/11379989/219547307-78301245-e397-4a93-b95a-b8e2d4b58097.png)
I get this scene as a result
![grafik](https://user-images.githubusercontent.com/11379989/219547270-15075d0b-f748-49f5-88d3-4421414d64bc.png)

MichiGandi committed:
fix deep copy scene

entity deep copy does already deep copy its children, so only root objects have to be deep copied.
